homeassistant/components/integration/sensor.py

```diff
@@ -254,6 +320,31 @@ def _unit(self, source_unit: str) -> str:
 
         return self._unit_template.format(integral_unit)
 
+    def _derive_and_set_attributes_from_state(self, source_state: State) -> None:
+        unit = source_state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+        if unit is not None:
+            self._unit_of_measurement = self._unit(unit)
+
+        if (
+            self.device_class is None
+            and source_state.attributes.get(ATTR_DEVICE_CLASS)
+            == SensorDeviceClass.POWER
+        ):
+            self._attr_device_class = SensorDeviceClass.ENERGY
+            self._attr_icon = None
```

Comment:
A comment would be nice here too IMHO

Reply:
About setting `self._attr_icon` to `None`? I don't know why this is/was done. I do know if this line is removed all unit tests still pass. Maybe we can remove it?